### PR TITLE
BUG: Fix build issues on SPR and avx512_qsort float16

### LINKED ISSUE #25274: BUG: Build failure (of 1.26.2) on SapphireRapids (`avx512_spr`) due to multiple definition of `avx512_qsort` and `avx512_qselect`
### Describe the issue:

Building 1.26.2 on SapphireRapids with
```
spin build -- -Dcpu-baseline=native
```
or, on IceLake with
```
spin build -- -Dcpu-baseline=avx512_spr
```

Fails due to multiple definition of `void avx512_qsort<_Float16>(_Float16*, long)` and `void avx512_qselect<_Float16>(_Float16*, long, long)`.

### Reproduce the code example:

```python
n/a
```

### Error message:

```shell
FAILED: numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so                                                                                                                                          c++  -o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_arraytypes.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_einsum.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_einsum_sumprod.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_lowlevel_strided_loops.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_nditer_templ.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_scalartypes.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_loops.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_matmul.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/meson-generated_scalarmath.c.o ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acos_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acos_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acos_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acosh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acosh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_acosh_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asin_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asin_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asin_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asinh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asinh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_asinh_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan2_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan2_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan2_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atan_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atanh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atanh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_atanh_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cbrt_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cbrt_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cbrt_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cos_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cos_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cos_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cosh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cosh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_cosh_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp2_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp2_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp2_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_exp_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_expm1_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_expm1_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_expm1_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log10_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log10_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log10_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log1p_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log1p_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log1p_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log2_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log2_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log2_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_log_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_pow_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_pow_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_pow_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sin_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sin_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sin_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sinh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sinh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_sinh_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tan_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tan_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tan_d_ha.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tanh_d_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tanh_s_la.s ../numpy/core/src/umath/svml/linux/avx512/svml_z0_tanh_d_ha.s numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_abstractdtypes.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_alloc.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_arrayobject.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_array_coercion.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_array_method.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_array_assign_scalar.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_array_assign_array.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_arrayfunction_override.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_buffer.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_calculation.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_compiled_base.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_common.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_common_dtype.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_convert.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_convert_datatype.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_conversion_utils.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_ctors.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_datetime.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_datetime_strings.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_datetime_busday.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_datetime_busdaycal.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_descriptor.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_dlpack.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_dtypemeta.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_dragon4.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_dtype_transfer.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_dtype_traversal.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_experimental_public_dtype_api.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_flagsobject.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_getset.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_hashdescr.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_item_selection.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_iterators.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_legacy_dtype_implementation.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_mapping.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_methods.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_multiarraymodule.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_nditer_api.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_nditer_constr.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_nditer_pywrap.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_number.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_refcount.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_sequence.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_shape.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_scalarapi.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_strfuncs.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_temp_elide.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_typeinfo.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_usertypes.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_vdot.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_quicksort.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_mergesort.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_timsort.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_heapsort.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_radixsort.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_selection.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npysort_binsearch.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_conversions.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_field_types.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_growth.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_readtext.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_rows.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_stream_pyobject.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_str_to_int.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_multiarray_textreading_tokenize.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_npymath_arm64_exports.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_array_assign.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_mem_overlap.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_npy_argparse.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_npy_hashtable.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_npy_longdouble.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_ucsnarrow.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_ufunc_override.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_numpyos.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_npy_cpu_features.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_cblasfuncs.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_common_python_xerbla.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_ufunc_type_resolution.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_clip.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_dispatching.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_extobj.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_legacy_array_method.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_override.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_reduction.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_ufunc_object.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_umathmodule.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_string_ufuncs.cpp.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_wrapping_array_method.c.o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath__scaled_float_dtype.c.o -Wl,--as-needed -Wl,--allow-shlib-undefined -shared -fPIC -Wl,--start-group numpy/core/libnpymath.a numpy/core/lib_multiarray_umath_mtargets.a /rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/OpenBLAS/0.3.24-GCC-13.2.0/lib/libopenblas.so -Wl,--end-group
/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/binutils/2.40-GCCcore-13.2.0/bin/ld: numpy/core/libsimd_qsort_16bit.dispatch.h_AVX512_SPR.a.p/src_npysort_simd_qsort_16bit.dispatch.cpp.o: in function `void avx512_qsort<_Float16>(_Float16*, long)':
/rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2/build/../numpy/core/src/npysort/x86-simd-sort/src/avx512fp16-16bit-qsort.hpp:161: multiple definition of `void avx512_qsort<_Float16>(_Float16*, long)'; numpy/core/libsimd_qsort_16bit.dispatch.h_AVX512_ICL.a.p/src_npysort_simd_qsort_16bit.dispatch.cpp.o:/rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2/build/../numpy/core/src/npysort/x86-simd-sort/src/avx512fp16-16bit-qsort.hpp:161: first defined here
/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/binutils/2.40-GCCcore-13.2.0/bin/ld: numpy/core/libsimd_qsort_16bit.dispatch.h_AVX512_SPR.a.p/src_npysort_simd_qsort_16bit.dispatch.cpp.o: in function `void avx512_qselect<_Float16>(_Float16*, long, long)':
/rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2/build/../numpy/core/src/npysort/x86-simd-sort/src/avx512fp16-16bit-qsort.hpp:149: multiple definition of `void avx512_qselect<_Float16>(_Float16*, long, long)'; numpy/core/libsimd_qsort_16bit.dispatch.h_AVX512_ICL.a.p/src_npysort_simd_qsort_16bit.dispatch.cpp.o:/rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2/build/../numpy/core/src/npysort/x86-simd-sort/src/avx512fp16-16bit-qsort.hpp:149: first defined here
collect2: error: ld returned 1 exit status
```

### Runtime information:

```
The Meson build system 
Version: 1.2.99
Source dir: /rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2
Build dir: /rds/projects/2017/branfosj-rse/ProblemSolving/numpy/numpy-1.26.2/build
Build type: native build
Project name: NumPy
Project version: 1.26.2
C compiler for the host machine: cc (gcc 13.2.0 "cc (GCC) 13.2.0")
C linker for the host machine: cc ld.bfd 2.40
C++ compiler for the host machine: c++ (gcc 13.2.0 "c++ (GCC) 13.2.0")
C++ linker for the host machine: c++ ld.bfd 2.40
Cython compiler for the host machine: cython (cython 3.0.4)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/Python/3.11.5-GCCcore-13.2.0/bin/python)
Found pkg-config: /rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/pkgconf/2.0.3-GCCcore-13.2.0/bin/pkg-config (2.0.3)
Run-time dependency python found: YES 3.11
Has header "Python.h" with dependency python-3.11: YES
Compiler for C supports arguments -fno-strict-aliasing: YES
Test features "SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL AVX512_SPR" : Supported
Test features "AVX512_KNL" : Supported
Test features "AVX512_KNM" : Supported
Configuring npy_cpu_dispatch_config.h using configuration
Message:
     CPU Optimization Options
baseline:
        Requested : avx512_spr
		Enabled   : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL AVX512_SPR
dispatch:
        Requested : max -xop -fma4
		Enabled   : AVX512_KNL AVX512_KNM
Library m found: YES
Run-time dependency scipy-openblas found: NO (tried pkgconfig)
Run-time dependency mkl found: NO (tried pkgconfig and system)
Run-time dependency mkl found: NO (tried pkgconfig and system)
Run-time dependency accelerate found: NO (tried system)
Run-time dependency openblas found: YES 0.3.24
Message: BLAS symbol suffix:
Run-time dependency mkl found: NO (tried pkgconfig and system)
Run-time dependency accelerate found: NO (tried system)
Run-time dependency openblas found: YES 0.3.24
```

And the last part of the configure
```
 Generating multi-targets for "_umath_tests.dispatch.h"                                                                                                                                                          Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "argfunc.dispatch.h"                                                                                                                                                               Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "simd_qsort.dispatch.h"                                                                                                                                                            Enabled targets: AVX512_SKX                                                                                                                                                                                 Generating multi-targets for "simd_qsort_16bit.dispatch.h"                                                                                                                                                      Enabled targets: AVX512_SPR, AVX512_ICL                                                                                                                                                                     Generating multi-targets for "loops_arithm_fp.dispatch.h"                                                                                                                                                       Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "loops_arithmetic.dispatch.h"                                                                                                                                                      Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "loops_comparison.dispatch.h"                                                                                                                                                      Enabled targets: baseline
Generating multi-targets for "loops_exponent_log.dispatch.h"                                                                                                                                                    Enabled targets: baseline
Generating multi-targets for "loops_hyperbolic.dispatch.h"                                                                                                                                                      Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "loops_logical.dispatch.h"
  Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "loops_minmax.dispatch.h"
  Enabled targets: baseline                                                                                                                                                                                   Generating multi-targets for "loops_modulo.dispatch.h"                                                                                                                                                          Enabled targets: baseline
Generating multi-targets for "loops_trigonometric.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_umath_fp.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_unary.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_unary_fp.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_unary_fp_le.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_unary_complex.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "loops_autovec.dispatch.h"
  Enabled targets: baseline
Generating multi-targets for "_simd.dispatch.h"
  Enabled targets: baseline
Build targets in project: 62

NumPy 1.26.2

  User defined options
    prefix      : /usr
    cpu-baseline: avx512_spr
```

### Context for the issue:

_No response_

### PR DESCRIPTION
**Pull request by r-devulap**
Fixes https://github.com/numpy/numpy/issues/25274, https://github.com/numpy/numpy/issues/25382

1. Marks explicit template specializations in x86-simd-sort as inline. 
2. Adds CI to run tests on TGL and SPR platforms with latest version of Intel SDE v9.27. 
3. Bug fix in `avx512_qsort<float16>`, which was detected when enabling the CI on TGL.

### CODE REVIEW

**Comment by r-devulap** [comment]
Enabling tests on SDE found a bug in avx512_qsort<float16>, which this patch fixes.

**Comment by seiko2plus** [comment]
It should also resolve #25382, Isn't?

**Comment by r-devulap** [comment]
> It should also resolve #25382, Isn't?

Yup, I found it when enabling the CI on Intel SDE TGL platform. It will be useful to run tests on TGL and SPR.

**Comment by r-devulap** [comment]
intel_sde CI takes about 35 min to complete, let me know if you want me to split it. We could run TGL and SPR tests in separate CI jobs.

**Comment by r-devulap** [comment]
`intel_sde` is split into `intel_sde_avx512` and `intel_sde_spr`

**Comment by r-devulap** [comment]
Something went wrong with the Qemu tests. Will close and open to re-run them.

File: .github/workflows/linux_simd.yml [hunk 1]
```diff
@@ -198,15 +198,59 @@ jobs:
         sde -spr -- python -c "import numpy; numpy.show_config()" &&
         sde -spr -- python -m pytest $NUMPY_SITE/numpy/_core/tests/test_simd*
```
Comment (seiko2plus) [inline .github/workflows/linux_simd.yml:199]:
```suggestion
```
There is no longer a need for this test, it would be better to move to the SPR job.

Comment (r-devulap) [reply .github/workflows/linux_simd.yml]: sure. Moved it to `inte_sde_spr` job.

Comment (seiko2plus) [reply .github/workflows/linux_simd.yml]: No, what I meant is to append `$NUMPY_SITE/numpy/_core/tests/test_simd*` to the action `- name: linalg/ufunc/umath tests on Intel SPR` to speed up the test.

Comment (r-devulap) [reply .github/workflows/linux_simd.yml]: oh ok. But this probably fine too, right? It's the same set of tests just running in a different job.

Comment (r-devulap) [reply .github/workflows/linux_simd.yml]: and the SIMD tests take only a min, when compared to the umath tests take like 12 min.

Comment (seiko2plus) [reply .github/workflows/linux_simd.yml]: Ok, lets put it in and try to optimize it later, thank you.

**Comment by seiko2plus** [comment]
Thank you Raghuveer.

**Comment by charris** [comment]
> Bug fix in avx512_qsort<float16>, which was detected when enabling the CI on TGL.

@r-devulap Is this bug present in 1.26.x? If so, could you make a bug fix PR for it? I don't want to try taking this whole PR back, the files have been renamed and reorganized.

**Comment by r-devulap** [comment]
@charris see https://github.com/numpy/numpy/pull/25475.

**Pull request merged**